security-analysis/security-analysis-api/src/main/java/com/powsybl/security/distributed/DistributedSecurityAnalysis.java

```diff
@@ -45,7 +45,7 @@ public DistributedSecurityAnalysis(ExternalSecurityAnalysisConfig config, Networ
 
         ExecutionEnvironment itoolsEnv = new ExecutionEnvironment(Collections.emptyMap(), "security_analysis_task_", config.isDebug());
 
-        int actualTaskCount = Math.min(taskCount, contingenciesProvider.getContingencies(network).size());
+        int actualTaskCount = Math.min(taskCount, contingenciesProvider.getContingencies(network).isEmpty() ? 1 : contingenciesProvider.getContingencies(network).size());
```

Comment:
As you don't know how `getContingencies()` is implemented, you should store the result in a temporary variable or call it once:
Solution 1:
```
List<Contingency> contingencies = contingenciesProvider.getContingencies(network);
int actualTaskCount = Math.min(taskCount, contingencies.isEmpty() ? 1 : contingencies.size());
```

Solution 2:
```
List<Contingency> contingencies = contingenciesProvider.getContingencies(network);
Math.min(taskCount, Math.max(1, contingencies.size());
```